pkg/controlplane/controller/leaderelection/leaderelection_controller.go

```diff
@@ -280,18 +284,21 @@ func (c *Controller) reconcileElectionStep(ctx context.Context, leaseNN types.Na
 			// If PingTime is outdated, send another PingTime only if it already acked the first one.
 			// This checks for pingTime <= renewTime because equality is possible in unit tests using a fake clock.
 			(candidate.Spec.PingTime.Add(electionDuration).Before(now) && !candidate.Spec.RenewTime.Before(candidate.Spec.PingTime)) {
-			// TODO(jefftree): We should randomize the order of sending pings and do them in parallel
-			// so that all candidates have equal opportunity to ack.
 			clone := candidate.DeepCopy()
 			clone.Spec.PingTime = &metav1.MicroTime{Time: now}
-			_, err := c.leaseCandidateClient.LeaseCandidates(clone.Namespace).Update(ctx, clone, metav1.UpdateOptions{})
-			if err != nil {
-				return defaultRequeueInterval, err
-			}
-			canVoteYet = false
+			g.Go(func() error {
+				if _, err := c.leaseCandidateClient.LeaseCandidates(clone.Namespace).Update(gCtx, clone, metav1.UpdateOptions{}); err != nil {
+					return err
+				}
+				canVoteYet.Store(false)
+				return nil
+			})
```

Comment:
Isn't this the same?
```suggestion
			g.Go(func() error {
				return c.leaseCandidateClient.LeaseCandidates(clone.Namespace).Update(gCtx, clone, metav1.UpdateOptions{})
			})
			canVoteYet = false
```

Reply:
In other words: `canVoteYet` becomes false when we go into line 276 or one of the go routines starts. If any of the go routines returns an error, we return it further down anyway and ignore `canVoteYet`.

Hence, no need for the atomic.

Reply:
> Isn't this the same?

c.leaseCandidateClient.LeaseCandidates().Update... 
return 2 values

thus can not directly return it to "func() error", now I just simply return err variable.

> In other words: canVoteYet becomes false ... no need for the atomic.
Yeah, thanks for the suggestion, that is more concise!